src/Aspire.Dashboard/Model/ResourceSourceViewModel.cs

```diff
@@ -31,46 +34,52 @@ public class ResourceSourceViewModel(string value, List<LaunchArgument>? content
                         .Select((arg, i) => new LaunchArgument(arg, IsShown: !areArgumentsSensitive[i]))
                         .ToList();
 
-                    commandLineInfo = (Arguments: arguments, argumentsString);
+                    commandLineInfo = new CommandLineInfo(
+                        Arguments: arguments,
+                        ArgumentsString: argumentsString,
+                        TooltipString: string.Join(" ", arguments.Select(arg => arg.IsShown
+                            ? arg.Value
+                            : WebUtility.HtmlDecode(DashboardUIHelpers.GetMaskingText(6).Value))));
```

Comment:
`WebUtility.HtmlDecode` here is a bit hacky. `GetMaskingText` was indented for text that's included in the Razor page. I'm guessing the decode is neccessary to avoid the already escaped characters being escaped again.

Perhaps `GetMaskingText` should return a record that has escaped and unescaped strings on it. You would then used the unescaped string here. Could do this as a follow up change.

Reply:
Sure, changed to a record, TextMask